hilti/src/compiler/driver.cc

```diff
@@ -730,11 +744,16 @@ Result<Nothing> Driver::outputUnits() {
     for ( auto& unit : _hlts ) {
         if ( auto cxx = unit.cxxCode() ) {
             if ( _driver_options.output_cxx ) {
-                auto output = openOutput(output_path, false);
+                auto cxx_path = output_path;
+
+                if ( _driver_options.output_cxx_prefix.size() )
```

Comment:
nit: prefer `empty` to check for emptiness.
```suggestion
                if ( ! _driver_options.output_cxx_prefix.empty() )
```

Do we need a similar check (or assert) of `cxx->id()` for non-emptiness? It could in principle be empty as well which could lead to collisions.

Reply:
I'll add an assert, if it's empty we're in trouble.

I need to add to the style guide that using `size()` instead if `! empty()` is acceptable. :)

Reply:
Well, in a way we already do by explicitly disabling [this](https://clang.llvm.org/extra/clang-tidy/checks/readability-container-size-empty.html) check in the clang-tidy config. But since the code is almost clean here we could alternatively enable the check to remove the annoying human from the equation ;)